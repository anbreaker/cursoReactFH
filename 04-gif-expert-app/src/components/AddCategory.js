import React, {useState} from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ({setCategories}) => {
  const [inputValue, setinputValue] = useState('');

  const handleInputChange = (event) => {
    setinputValue(event.target.value);
  };

  const handleSubmit = (event) => {
    event.preventDefault();
    if (inputValue.trim().length > 2) {
      setCategories((categories) => [inputValue, ...categories]);
      setinputValue('');
    }
  };

  return (
    <form onSubmit={handleSubmit}>
      <input
        type='text'
        value={inputValue}
        onChange={handleInputChange}
        placeholder='Search'
      />
    </form>
  );
};

AddCategory.propTypes = {
  setCategories: PropTypes.func.isRequired,
};
