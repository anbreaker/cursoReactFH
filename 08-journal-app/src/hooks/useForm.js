import {useState} from 'react';

export const useForm = (initialStateForm = {}) => {
  const [formValues, setValues] = useState(initialStateForm);

  const reset = (newFormState = initialStateForm) => {
    setValues(newFormState);
  };

  const handleInputChange = ({target}) => {
    setValues({
      ...formValues,
      [target.name]: target.value,
    });
  };

  return [formValues, handleInputChange, reset];
};
