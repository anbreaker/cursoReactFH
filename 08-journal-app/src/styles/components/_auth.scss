.auth__main {
  align-items: center;
  background-color: $primary;
  display: flex;
  justify-content: center;
  margin: 0px;
  height: 100vh;
  width: 100vw;
}

.auth__box-container {
  background-color: azure;
  box-shadow: 0px 3px $dark-grey;
  border-radius: 2px;
  padding: 20px;
  width: 250px;
}

.auth__title {
  color: darken($color: $primary, $amount: 20);
  margin-bottom: 20px;
}

.auth__input {
  border-bottom: 1px solid $light-grey;
  border: 0px;
  color: $dark-grey;
  font-size: 16px;
  height: 20px;
  margin-bottom: 10px;
  width: 100%;

  transition: border-bottom 0.3s ease;

  &:focus {
    border-bottom: 1px solid $primary;
    outline: none;
  }
}

.auth__social-networks {
  align-items: center;
  display: flex;
  flex-direction: column;
  justify-content: center;
  padding-bottom: 20px;
  padding-top: 20px;
  width: 100%;
}

.auth__alert-error {
  background-color: red;
  border-radius: 5px;
  color: azure;
  display: flex;
  justify-content: center;
  margin-bottom: 10px;
  padding: 5px;
}
