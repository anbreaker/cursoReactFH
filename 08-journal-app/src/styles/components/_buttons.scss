.btn {
  background-color: transparent;
  border: none;
  color: azure;
  cursor: pointer;
  font-size: 12px;
  padding: 7px 10px 7px 10px;

  transition: color 0.3s ease;

  &:focus {
    outline: none;
  }

  &:hover {
    color: darken($color: azure, $amount: 10);
  }
}

.btn-primary {
  background-color: $primary;
  border-radius: 2px;

  &:disabled {
    background-color: lighten($color: $primary, $amount: 1);
  }

  &:hover {
    background-color: darken($color: $primary, $amount: 15);
  }
}

.btn-block {
  width: 100%;
}

.btn-danger {
  background-color: crimson;
  transition: background-color 0.3s;

  &:hover {
    background-color: darken($color: crimson, $amount: 10);
  }
}

.google-btn {
  background-color: $google-blue;
  border-radius: 2px;
  box-shadow: 0 3px 4px 0 rgba(0, 0, 0, 0.25);
  cursor: pointer;
  height: 42px;
  margin-top: 5px;
  width: 100%;

  transition: box-shadow 0.3s ease;

  .google-icon-wrapper {
    background-color: $white;
    border-radius: 2px;
    height: 40px;
    margin-left: 1px;
    margin-top: 1px;
    position: absolute;
    width: 40px;
  }

  .google-icon {
    height: 18px;
    margin-left: 11px;
    margin-top: 11px;
    position: absolute;
    width: 18px;
  }

  .btn-text {
    color: $white;
    float: right;
    font-size: 14px;
    letter-spacing: 0.2px;
    margin: 11px 40px 0 0;
  }

  &:hover {
    box-shadow: 0 0 6px $google-blue;
  }

  &:active {
    background: $button-active-blue;
  }
}
