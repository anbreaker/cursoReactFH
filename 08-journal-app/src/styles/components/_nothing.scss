.nothing__main-content {
  align-items: center;
  background-color: $primary;
  color: azure;
  display: flex;
  flex-direction: column;
  height: 100vh;
  justify-content: center;

  p {
    text-align: center;
  }
}
