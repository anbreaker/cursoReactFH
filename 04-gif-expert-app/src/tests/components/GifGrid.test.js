import React from 'react';
import {shallow} from 'enzyme';
import {GifGrid} from '../../components/GifGrid';
import '@testing-library/jest-dom';
import {useFetchGifs} from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('Pruebas en el <GifGrid /> ', () => {
  const category = 'Dragon ball';

  test('debe mostrarse correctamente', () => {
    useFetchGifs.mockReturnValue({
      data: [],
      loading: true,
    });

    const wrapper = shallow(<GifGrid category={category} />);
    expect(wrapper).toMatchSnapshot();
  });

  test('debe de mostrar items cuando se cargan imagenes useFetchGifs', () => {
    const gifs = [
      {
        id: 'ABC',
        url: 'https://localhost/cualquier/cosa.gif',
        title: 'Cualquier cosa',
      },
      {
        id: '123',
        url: 'https://localhost/cualquier/cosa.gif',
        title: 'Cualquier cosa',
      },
    ];
    useFetchGifs.mockReturnValue({
      data: gifs,
      loading: false,
    });

    const wrapper = shallow(<GifGrid category={category} />);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('p').exists()).toBe(false);
    expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
  });
});
