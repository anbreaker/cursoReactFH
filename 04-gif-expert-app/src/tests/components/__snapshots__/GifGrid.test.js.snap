// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el <GifGrid />  debe de mostrar items cuando se cargan imagenes useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Dragon ball
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="Cualquier cosa"
      url="https://localhost/cualquier/cosa.gif"
    />
    <GifGridItem
      id="123"
      key="123"
      title="Cualquier cosa"
      url="https://localhost/cualquier/cosa.gif"
    />
  </div>
</Fragment>
`;

exports[`Pruebas en el <GifGrid />  debe mostrarse correctamente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Dragon ball
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
